frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -387,19 +388,23 @@ def _callee():
 
 
 ## IMPL ##
-class MidCircuitMeasure(HybridOp):
+class MidCircuitMeasure(HybridOp, MidMeasureMP):
     """Operation representing a mid-circuit measurement."""
 
     binder = qmeasure_p.bind
 
+    def __init__(self, in_classical_tracers, out_classical_tracers, regions: List[HybridOpRegion]):
+        HybridOp.__init__(self, in_classical_tracers, out_classical_tracers, regions)
+        # Initializing MidMeasureMP attributes
+        self.wires = [in_classical_tracers[0]]
+        self.postselect = in_classical_tracers[1]
+        # TODO: Check if we need to initialize other MidMeasureMP attributes as well
```

Comment:
I think it might also be important for the one-shot that we link back to the measurement value (in this case `out_classical_tracers[0]`).

Reply:
Okay, I will add a new attribute for that, as `MidMeasureMP` doesn't have any attributes to link back to the measurement values. @vincentmr this may be useful for your work.

Reply:
If it's not too difficult, could you rebase and merge into #773 . This way you could also use the `dynamic_one_shot` unit tests.

Reply:
> Okay, I will add a new attribute for that, as MidMeasureMP doesn't have any attributes to link back to the measurement values. @vincentmr this may be useful for your work.

I thought it did via the `.mv` property, am I mixing something up?

Reply:
No. All `MeasurementProcess`s have a `mv` attribute, but it's only used by terminal measurements. `MidMeasureMP` does not do anything with `mv`.

Reply:
Oh I see, but it is always used to hold references to `MeasurementValue` (i.e. the result of a `MidMeasureMP`) objects right?

In that case,  `MidMeasureMP` could hold a reference to its own result just the same, or this is a bad idea for some reason?